src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java

```diff
@@ -232,9 +241,15 @@ public void remove(Graph graph) {
                         }
                     }
 
-                    for (int i = 0; i < tripPattern.hopEdges.length; i++) {
-                        if (stop == null || stop.equals(tripPattern.hopEdges[i].getEndStop())) {
-                            graph.removeAlertPatch(tripPattern.hopEdges[i], this);
+                    for (int i = 0; i < tripPattern.dwellEdges.length; i++) {
+                        PatternDwell dwellEdge = tripPattern.dwellEdges[i];
+
+                        if (dwellEdge != null) {
+                            Stop fromv = ((PatternArriveVertex) dwellEdge.getFromVertex()).getStop();
+                            Stop tov = ((PatternDepartVertex) dwellEdge.getToVertex()).getStop();
+                            if (stop == null || (stop.equals(fromv) && stop.equals(tov))) {
```

Comment:
Is it possible for the the stop to equal both `fromv` and `tov`? It seems like those would be separate things.

Reply:
Well, those are different vertexes (`PatternArriveVertex` and `PatternDepartVertex`), but they are both linked to the same transit stop. It sounds a bit counter-intuitive, but this is about `PatternDwell` edges, which are used to model _waiting_ on the stop in a transit vehicle, so it kind of makes sense that both ends of such an edge belong to the same stop. 

<img width="716" alt="image" src="https://user-images.githubusercontent.com/45011335/57697174-a32a4c00-7620-11e9-9f11-44135ed13d06.png">

Reply:
Okay, so it's validating that the fromv, tov, and stop are all the same?

Reply:
Yes, this is what it is doing.